### DISCUSSION TITLE
not a valid keg for zlib in Big Sur

### DISCUSSION THREAD

**Discussion by aotarola** [Homebrew/discussions#338] [Everyday usage] [answered]
Hello!, after a clean install of Big Sur, I installed homebrew and tried to install `zlib` but failed 😭 . I have yet to see anyone having the same issue, so it might not be widespread? 🤷🏼 

```sh
$ brew install --verbose --debug zlib
Updating Homebrew...
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/zlib.rb
Error: /usr/local/opt/zlib is not a valid keg
/usr/local/Homebrew/Library/Homebrew/keg.rb:224:in `initialize'
/usr/local/Homebrew/Library/Homebrew/cmd/install.rb:283:in `new'
/usr/local/Homebrew/Library/Homebrew/cmd/install.rb:283:in `block in install'
/usr/local/Homebrew/Library/Homebrew/cmd/install.rb:175:in `each'
/usr/local/Homebrew/Library/Homebrew/cmd/install.rb:175:in `install'
/usr/local/Homebrew/Library/Homebrew/brew.rb:119:in `<main>'
```

doctor seems happy tho 👨🏼‍⚕️ 
```sh
$ brew doctor
Your system is ready to brew.
```

Any ideas on what to try? 🤔

**Comment by gromgit** [comment, marked as answer]
What's the output of `ls -ld /usr/local/opt /usr/local/opt/zlib`?

**Reply by aotarola** [reply]
```sh
 $ ls -ld /usr/local/opt /usr/local/opt/zlib
drwxrwxr-x  43 aotarola  admin  1376 Dec 21 00:20 /usr/local/opt
drwxr-xr-x   3 aotarola  admin    96 Dec 20 21:52 /usr/local/opt/zlib
```

**Reply by gromgit** [reply]
> ```shell
> drwxr-xr-x   3 aotarola  admin    96 Dec 20 21:52 /usr/local/opt/zlib
> ```

There's your problem: Everything under `/usr/local/opt` (for now, at least) should be symlinks to Cellar directories:
```
lrwxr-xr-x 1 aho admin 21 Dec 18 01:56 /usr/local/opt/zlib -> ../Cellar/zlib/1.2.11
```
Since Homebrew wouldn't do that, you might want to check the contents of `/usr/local/opt/zlib`, before deleting it entirely and `brew install zlib` again.

**Reply by aotarola** [reply]
That did the trick!. Thanks for your fast reply and resolution 👍🏼
